public_html/wp-content/mu-plugins/latest-site-hints.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace WordCamp\Latest_Site_Hints;
+use function WordCamp\Sunrise\get_top_level_domain;
+
+use const WordCamp\Sunrise\{ PATTERN_YEAR_DOT_CITY_DOMAIN_PATH, PATTERN_CITY_SLASH_YEAR_DOMAIN_PATH };
+
+defined( 'WPINC' ) || die();
+
+add_action( 'wp', __NAMESPACE__ . '\maybe_add_latest_site_hints' );
+
+/**
+ * If user or bot visits WordCamp site that has newer site for the same city,
+ * add some hints for guiding them visit the latest site.
+ */
+function maybe_add_latest_site_hints() {
+	// Check if there is newer site for the WordCamp.
+	$latest_site = get_latest_site();
+	if ( ! $latest_site ) {
+		return;
+	}
+
+	// Hook in before `WordPressdotorg\SEO\Canonical::rel_canonical_link()`, so that callback can be removed.
+	add_action( 'wp_head', __NAMESPACE__ . '\canonical_link_past_home_pages_to_current_year', 9 );
+
+	/**
+	 * Show notification about newer WordCamp. Logged in users most probably know that already,
+	 * so no need to bother them (and we also misuse the body.admin-bar class).
+	 */
+	if ( ! is_user_logged_in() ) {
+		add_filter( 'body_class',   __NAMESPACE__ . '\add_notification_classes_to_body' );
+		add_action( 'wp_head',      __NAMESPACE__ . '\add_notification_styles' );
+		add_action( 'wp_body_open', __NAMESPACE__ . '\show_notification_about_latest_site' );
```

Comment:
This hook is only called if the theme uses [the `wp_body_open` function](https://make.wordpress.org/core/2019/04/24/miscellaneous-developer-updates-in-5-2/), and CampSite 2017 doesn't — if there isn't another suitable hook, the theme can be updated.

Reply:
Oh, good catch! Changed the message output to `wp_footer`, but I'm not too thrilled about that because of accessibility. 🤔

Reply:
> This hook is only called if the theme uses [the wp_body_open function](https://make.wordpress.org/core/2019/04/24/miscellaneous-developer-updates-in-5-2/), and CampSite 2017 doesn't

Regardless, we should probably add that to the theme just for future compatibility.

Reply:
Agreed. I will open a PR on the theme, and let's switch to `wp_body_open` when that is in production.